moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group_interface.h

```diff
@@ -362,7 +361,7 @@ class MoveGroupInterface
 
       If these values are out of bounds then false is returned BUT THE VALUES
       ARE STILL SET AS THE GOAL. */
-  bool setJointValueTarget(const robot_state::RobotState& robot_state);
+  MOVEIT_DEPRECATED bool setJointValueTarget(const robot_state::RobotState& robot_state);
```

Comment:
I agree that these functions should not set any positions other than those inside the corresponding joint model group.
Rather than making these deprecated I would really prefer changing the implementation to just use `RobotState::copyJointGroupPositions()` or `RobotState::getVariablePositions()` selectively.

Reply:
I my opinion, the RobotState shouldn't be exposed by move-group methods.

Reply:
In that case `setStartState(robot_state::RobotState)` should be deprecated as well. But does this really improve usability?